src/test/java/de/tum/in/www1/artemis/service/UrlServiceTest.java

```diff
@@ -9,49 +9,92 @@
 
 import de.tum.in.www1.artemis.AbstractSpringIntegrationBambooBitbucketJiraTest;
 import de.tum.in.www1.artemis.domain.VcsRepositoryUrl;
+import de.tum.in.www1.artemis.domain.participation.ProgrammingExerciseStudentParticipation;
+import de.tum.in.www1.artemis.domain.participation.SolutionProgrammingExerciseParticipation;
+import de.tum.in.www1.artemis.domain.participation.TemplateProgrammingExerciseParticipation;
 import de.tum.in.www1.artemis.exception.VersionControlException;
 
 public class UrlServiceTest extends AbstractSpringIntegrationBambooBitbucketJiraTest {
 
+    private final VcsRepositoryUrl repositoryUrl1 = new VcsRepositoryUrl("https://ga42xab@bitbucket.ase.in.tum.de/scm/EIST2016RME/RMEXERCISE-ga42xab");
+
+    private final VcsRepositoryUrl repositoryUrl2 = new VcsRepositoryUrl("https://ga63fup@repobruegge.in.tum.de/scm/EIST2016RME/RMEXERCISE-ga63fup.git");
+
+    private final VcsRepositoryUrl repositoryUrl3 = new VcsRepositoryUrl("https://artemistest2gitlab.ase.in.tum.de/TESTADAPTER/testadapter-exercise.git");
+
+    private final VcsRepositoryUrl repositoryUrl4 = new VcsRepositoryUrl("https://turdiu@artemistest2gitlab.ase.in.tum.de/FTCSCAGRADING1/ftcscagrading1-turdiu");
+
+    /**
+     * empty constructor for exception handling
+     * @throws MalformedURLException exception in case the above URLs would be malformed
+     */
+    public UrlServiceTest() throws MalformedURLException {
+    }
```

Comment:
Maybe it would be better to move the initialization of the repo URLs into the constructor?
This should get rid of the codacy issue and we don't need the comment explaining the empty constructor.